File: API/models/direccion.py
from typing import Optional
from pydantic import BaseModel

class DireccionBase(BaseModel):
    calle_principal: str
    num_ext: str
    num_int: Optional[str] = None
    colonia: str
    municipio: str
    estado: str
    cp: str
    referencias: Optional[str] = None
    cliente_id: Optional[int] = None

File: Laravel/app/Models/Macuin/Direccion.php
<?php

namespace App\Models\Macuin;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Direccion extends Model
{
    protected $table = 'direcciones';

    public $timestamps = false;

    protected $fillable = [
        'calle_principal',
        'num_ext',
        'num_int',
        'colonia',
        'municipio',
        'estado',
        'cp',
        'referencias',
        'cliente_id',
    ];

    public function cliente(): BelongsTo
    {
        return $this->belongsTo(Cliente::class, 'cliente_id');
    }
}
